src/components/pagination/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { FC, PropsWithChildren, useMemo } from 'react';
+import { useMediaQuery, useMultiStyleConfig } from '@chakra-ui/react';
+
+import PaginationButton from './PaginationButton';
+import { ChevronLeftSmallIcon, ChevronRightSmallIcon } from '../icons';
+import Box from '../box';
+import { breakpoints } from '../../themes';
+
+const Dots = '...';
+const siblingCount = 1;
+const range = (start: number, end: number): Array<number> => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+export interface Props {
+  totalPages: number;
+  currentPage: number;
+  onChange: (page: number) => void;
+}
+
+const Pagination: FC<PropsWithChildren<Props>> = (props) => {
+  const { onChange, totalPages, currentPage } = props;
+  const { paginationContainer, dots } = useMultiStyleConfig('Pagination');
+  const [isLargerThanXs] = useMediaQuery(`(min-width: ${breakpoints.xs.px}px)`);
+
+  const paginationRange = useMemo(() => {
+    // Default number of page buttons: firstPage + lastPage + currentPage + left side dots + right side dots
+    const pageButtonsCount = 5;
+    const totalPageButtonsCount = siblingCount + pageButtonsCount;
+
+    if (totalPageButtonsCount >= totalPages) {
+      return range(1, totalPages);
+    }
+
+    const leftSiblingIndex = Math.max(currentPage - siblingCount, 1);
+    const rightSiblingIndex = Math.min(currentPage + siblingCount, totalPages);
+
+    const shouldShowLeftDots = leftSiblingIndex > 2;
+    const shouldShowRightDots = rightSiblingIndex < totalPages - 2;
+
+    const firstPageIndex = 1;
```

Comment:
```suggestion
    const firstPageNumber = 1;
```
because the index would be 0 and not 1, right?